core/domain/suggestion_registry.py

```diff
@@ -680,14 +682,46 @@ def __init__(
         self.target_version_at_submission = target_version_at_submission
         self.author_id = author_id
         self.change = question_domain.QuestionSuggestionChange(change)
-        # Update question_state_data_schema_version here instead of surfacing
-        # the version in the frontend.
-        self.change.question_dict['question_state_data_schema_version'] = (
-            feconf.CURRENT_STATE_SCHEMA_VERSION)
         self.score_category = score_category
         self.language_code = language_code
         self.last_updated = last_updated
         self.image_context = feconf.IMAGE_CONTEXT_QUESTION_SUGGESTIONS
+        self._updated_change()
+
+    def _updated_change(self):
+        """Holds the responsibility of performing a step-by-step, sequential
+        update of the state structure based on the schema version of the current
+        state dictionary.
+
+        Raises:
+            Exception. The given state_schema_version is invalid.
+        """
+        state_schema_version = self.change.question_dict[
+            'question_state_data_schema_version']
+
+        versioned_question_state = {
+            'state': copy.deepcopy(
+                self.change.question_dict['question_state_data'])
+        }
+
+        if state_schema_version is None or state_schema_version < 1:
+            state_schema_version = 0
+
+        if not (25 <= state_schema_version
+                <= feconf.CURRENT_STATE_SCHEMA_VERSION):
+            raise Exception(
+                'Sorry, we can only process v25-v%d state schemas at present.' %
```

Comment:
Users aren't aware of the state schemas, rewrite it like our genre Exception messages*

Reply:
Done!